File: backend/fookbase.API/Src/Domain/Entities/Notification.cs
using InteractHub.Api.Domain.Enums;

namespace InteractHub.Api.Domain.Entities;

public class Notification
{
    public Guid Id { get; set; }

    public Guid UserId { get; set; }

    public Guid ActorUserId { get; set; }

    public Guid? PostId { get; set; }

    public Guid? CommentId { get; set; }

    public Guid? StoryId { get; set; }

    public NotificationType Type { get; set; }

    public string Message { get; set; } = string.Empty;

    public bool IsRead { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }
}

File: frontend/src/interface/notification.ts
export interface NotificationItem {
  id: string;
  message: string;
  createdAt: string;
  isRead: boolean;
  type?: string;
  actorUserId?: string;
  postId?: string;
  commentId?: string;
  avatarUrl?: string;
  requestId?: string;
  isVirtual?: boolean;
}

export interface NotificationPayload {
  id: string;
  userId: string;
  actorUserId: string;
  postId?: string | null;
  commentId?: string | null;
  type?: string | null;
  message: string;
  isRead: boolean;
  createdAt: string;
}

